alphastats/llm/uniprot_utils.py

```diff
@@ -1,132 +1,92 @@
-    "go",
-    "go_p",
-    "go_c",
-    "go_f",
-    # Pathology & Biotech
-    "cc_disease",
-    "cc_disruption_phenotype",
-    "cc_pharmaceutical",
-    "ft_mutagen",
-    "ft_act_site",
-    # Structure
-    "cc_subcellular_location",
-    "organelle",
-    "absorption",
-    # Publications
-    "lit_pubmed_id",
-    # Family & Domains
-    "protein_families",
-    "cc_domain",
-    "ft_domain",
-    # Protein-Protein Interaction Databases
-    "xref_biogrid",
-    "xref_intact",
-    "xref_mint",
-    "xref_string",
-    # Chemistry Databases
-    "xref_drugbank",
-    "xref_chembl",
-    "reviewed",
-]
-
-
-def get_uniprot_data(
-    gene_name: str,
-    organism_id: str,
-    fields: List[str] = uniprot_fields,
-) -> Dict:
+
+from alphastats.dataset.keys import ConstantsClass
+
+
+class ExtractedFields(metaclass=ConstantsClass):
+    DB = "entryType"
+    ID = "primaryAccession"
+    SECONDARYACC = "secondaryAccessions"
+    NAME = "protein"
+    GENE = "genes"
+    FUNCTIONCOMM = "functionComments"
+    SUBUNITCOMM = "subunitComments"
+    CAUTIONCOMM = "cautionComments"
+    INTERACTIONS = "interactions"
+    SUBCELL = "subcellularLocations"
+    TISSUE = "tissueSpecificity"
+    GOP = "GO Pathway"
+    GOF = "GO Function"
+    GOC = "GO Component"
+    REACTOME = "Reactome"
+
+
+def _request_uniprot_data(
+    protein_id: str = None,
+    gene_name: str = None,
+    organism_id: str = "9606",
+) -> Union[List, None]:
     """
-    Get data from UniProt for a given gene name and organism ID.
+    Get data from UniProt for a given gene name and organism ID, or for a specific Uniprot identifier.
 
     Args:
+        protein_id (str): Uniprot identifier of a protein
         gene_name (str): The gene name to search for.
-        organism_id (str, optional): The organism ID to search in. Defaults to streamlit session state.
-        fields (list[str], optional): The fields to retrieve from UniProt. Defaults to uniprot_fields defined above.
+        organism_id (str, optional): The organism ID to search in. Defaults to human, only used in the context of a gene name.
 
     Returns:
-        dict: The data retrieved from UniProt.
+        list: The data retrieved from UniProt, if retrieval was successful. If the response code was not 200 or there were no results the list is empty.
     """
     base_url = "https://rest.uniprot.org/uniprotkb/search"
-    query = f"(gene:{gene_name}) AND (reviewed:true) AND (organism_id:{organism_id})"
+    if protein_id is not None:
+        query = f"accession:{protein_id}"
+    elif gene_name is not None:
+        query = f"(gene:{gene_name}) AND (organism_id:{organism_id})"
+    else:
+        raise ValueError("Please provide either protein id or gene name.")
 
-    response = requests.get(
-        base_url, params={"query": query, "format": "json", "fields": ",".join(fields)}
-    )
+    response = requests.get(base_url, params={"query": query, "format": "json"})
 
     if response.status_code != 200:
```

Comment:
FYI for the test data set I retrieve some logs like 
```
Failed to retrieve data for accession:CON__P00735. Status code: 400
{"url":"http://rest.uniprot.org/uniprotkb/search","messages":["The 'accession' filter value 'CON__P00735' has invalid format. It should be a valid UniProtKB accession"]}
Failed to retrieve data for accession:CON__P28800. Status code: 400
{"url":"http://rest.uniprot.org/uniprotkb/search","messages":["The 'accession' filter value 'CON__P28800' has invalid format. It should be a valid UniProtKB accession"]}
Failed to retrieve data for accession:CON__ENSEMBL:ENSBTAP00000016046. Status code: 400
{"url":"http://rest.uniprot.org/uniprotkb/search","messages":["query parameter has an invalid syntax"]}
```

Reply:
Yes, that is fine. These are contaminants encoded in a MQ specific way and are not supposed to yield a result. If a feature is only this id, no uniprot data will be available for it.